assets/js/components/ChartSourceController/components/BarChart/services/createChartJsConfig/index.js

```diff
@@ -10,214 +10,248 @@ import isObjectLiteral from './services/isObjectLiteral/index.js';
+    const result = pivotedValues.map(formatDataset({color, barTypes}), []);
+    return result;
 };
 
 
-const createChartJsConfig = ({ items, rotated, color, viewportWidth, barTypes }) => {
-  const { labels, values } = flattenNesting(items);
-  const rotateLabels = viewportWidth && viewportWidth < 600 && rotated;
-  const datasets = buildDatasets(barTypes, values, color);
-
-  return {
-    type: rotated ? 'bar' : 'horizontalBar',
-    data: {
-      labels,
-      datasets,
-    },
-    options: {
-      barThickness: 10,
-      maintainAspectRatio: false,
-      tooltips: {
-        intersect: rotated,
-        custom: (tooltip) => {
-          if (!tooltip || /(^heading:\s)(.+)/im.test(tooltip.title)) {
-            tooltip.opacity = 0;
-            return;
-          }
-          tooltip.displayColors = false;
-        },
-        callbacks: {
-          label: (item, dataObject) => {
-            const { index, datasetIndex } = item;
-            const { data, label } = dataObject.datasets[datasetIndex];
-            const prefix = barTypes ? `${label}: ` : '';
-
-            return `${prefix}R${trimValues(data[index])}`;
-          },
+const createChartJsConfig = ({items, rotated, color, viewportWidth, barTypes}) => {
+    let {labels, values} = flattenNesting(items);
+    const rotateLabels = viewportWidth && viewportWidth < 600 && rotated;
+    let datasets = buildDatasets(barTypes, values, color);
+
+    return {
+        type: rotated ? 'bar' : 'horizontalBar',
+        data: {
+            labels,
+            datasets,
         },
-      },
-      animation: {
-        duration: 0,
-      },
-      layout: {
-        padding: {
-          top: 15,
-          bottom: 15,
+        options: {
+            barThickness: 10,
+            maintainAspectRatio: false,
+            tooltips: {
+                intersect: rotated,
+                custom: (tooltip) => {
+                    if (!tooltip || /(^heading:\s)(.+)/im.test(tooltip.title)) {
+                        tooltip.opacity = 0;
+                        return;
+                    }
+                    tooltip.displayColors = false;
+                },
+                callbacks: {
+                    label: (item, dataObject) => {
+                        const {index, datasetIndex} = item;
+                        const {data, label} = dataObject.datasets[datasetIndex];
+                        const prefix = barTypes ? `${label}: ` : '';
+
+                        return `${prefix}R${trimValues(data[index])}`;
+                    },
+                    title: (data, a, b) => {
+                        if (data.length <= 0) {
+                            return;
+                        }
+                        if (data[0].datasetIndex <= 3) {
+                            return `${data[0].label}`
+                        } else {
+                            return `${data[0].label} Q${data[0].datasetIndex - 3}`
+                        }
+                    }
```

Comment:
this index-based stuff will work even if Q1 is nonzero, Q2 is zero and Q3 is nonzero because all quarters always exist in the chart data, right?

Reply:
I just tested this by hardcoding Q2 value to 0 and it worked